### PR TITLE
Getting ready for npm publishing

### PR DESCRIPTION
**Pull request by dancrumb**
Making some changes to support publishing to NPM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,15 +4,20 @@
...
+    "url": "git+https://github.com/jpaulm/jsfbp.git"
+  },
+  "engines": {
+    "node" : "~4.3.1"
   },
   "keywords": [
     "fbp",
     "flow-based-programming",
-    "node-fibers",
-    "javascript"
```
Comment (dancrumb) [inline package.json]: Seems redundant, since npm is just for node modules

File: package.json [hunk 2]
```diff
@@ -4,15 +4,20 @@
...
+    "node-fibers"
   ],
   "author": "J. Paul Morrison",
+  "contributors": [
```
Comment (dancrumb) [inline package.json]: Let me know if you'd rather not have this

File: package.json [hunk 3]
```diff
@@ -23,15 +28,13 @@
...
     "mocha": "^2.1.0",
     "mocha-fibers": "^1.1.0",
-    "ws": "^0.7.1"
+    "ws": "^1.1.0"
   },
   "directories": {
-    "doc": "docs",
-    "example": "examples",
-    "test": "test"
+    "doc": "docs"
```
Comment (dancrumb) [inline package.json]: This is the only valid field of the 3 that you had, per https://docs.npmjs.com/files/package.json#directories

**Comment by jpaulm** [comment]
I just had to upgrade posh-git (I think?) - and I'm getting a message from
git pull:

C:\Users\Paul\Documents\GitHub\jsfbp [master =]> git pull
error: cannot open .git/FETCH_HEAD: Permission denied

Any idea what's wrong?  TIA

**Comment by dancrumb** [comment]
I'm guessing you're using PowerShell, which is a shell I'm not familiar with.

In a pure Unix environment, I would suggest simply checking the permission and ownership of the `.git` directory. Does PowerShell maintain and respect permissions?

**Comment by jpaulm** [comment]
Using Win7...  it has something to do with permissions, but I don't have
time to chase it right now!

Regards,

Paul

**Pull request merged**